core/js/src/Instant.kt

```diff
@@ -111,13 +111,12 @@ public actual fun Instant.plus(period: DateTimePeriod, timeZone: TimeZone): Inst
     val thisZdt = this.value.atZone(timeZone.zoneId)
     with(period) {
         thisZdt
-                .run { if (years != 0 && months == 0) plusYears(years) else this }
-                .run { if (months != 0) plusMonths(years * 12.0 + months) else this }
+                .run { if (totalMonths != 0) plusMonths(totalMonths) else this }
```

Comment:
There could have been a reason why years were not fused with months before when months were 0.

Reply:
I'm assuming we did it because it's slightly more efficient due to not using division: https://github.com/ThreeTen/threetenbp/blob/master/src/main/java/org/threeten/bp/LocalDate.java#L1105-L1142 (the js-joda implementation is virtually the same). With the orthogonal implementation, getting the number of years and months performs this division anyway.

Reply:
ok, makes sense.